### PR TITLE
osd,messages: changes for preparing for crimson-osd

### PR DESCRIPTION
**Pull request by tchaikov**
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by tchaikov** [comment]
this change is part of #26697

File: src/osd/OSD.cc [hunk 1]
```diff
@@ -9478,7 +9478,7 @@ void OSD::handle_pg_query_nopg(const MQuery& q)
 	    osdmap->get_epoch(),
 	    empty),
 	  PastIntervals()));
-      m = new MOSDPGNotify(osdmap->get_epoch(), ls);
+      m = new MOSDPGNotify(osdmap->get_epoch(), std::move(ls));
```
Comment (rzarzynski) [inline src/osd/OSD.cc]: Moving here looks fine.

File: src/osd/OSD.cc [hunk 2]
```diff
@@ -9184,7 +9184,7 @@ void OSD::do_notifies(
     dout(7) << __func__ << " osd." << it->first
 	    << " on " << it->second.size() << " PGs" << dendl;
     MOSDPGNotify *m = new MOSDPGNotify(curmap->get_epoch(),
-				       it->second);
+				       std::move(it->second));
```
Comment (rzarzynski) [inline src/osd/OSD.cc]: Basically this is about `notify_list` taken by non-const reference. `OSD::do_notifies` can be called only from `OSD::dispatch_context` which is turn has 3 users in `OSD.cc`. In each of them `PG::RecoveryCtx` are freshly created.

Also the two `MOSDPGNotify` instances are brand-new, so their `pg_list` members must be empty. In consequence there shouldn't be a difference between `swap` and `move`.

 Seems to be fine.

File: src/osd/OSD.cc [hunk 3]
```diff
@@ -9213,7 +9213,8 @@ void OSD::do_queries(map<int, map<spg_t,pg_query_t> >& query_map,
     service.share_map_peer(who, con.get(), curmap);
     dout(7) << __func__ << " querying osd." << who
 	    << " on " << pit->second.size() << " PGs" << dendl;
-    MOSDPGQuery *m = new MOSDPGQuery(curmap->get_epoch(), pit->second);
+    MOSDPGQuery *m = new MOSDPGQuery(curmap->get_epoch(),
```
Comment (rzarzynski) [inline src/osd/OSD.cc]: OK.

File: src/osd/PG.cc [hunk 4]
```diff
@@ -9180,7 +9180,7 @@ boost::statechart::result PG::RecoveryState::GetLog::react(const MLogRec& logevt
 		       << "non-auth_log_shard osd." << logevt.from << dendl;
     return discard_event();
   }
-  ldout(pg->cct, 10) << "GetLog: received master log from osd"
+  ldout(pg->cct, 10) << "GetLog: received master log from osd."
```
Comment (rzarzynski) [inline src/osd/PG.cc]: Good catch. Thanks!

File: src/messages/MOSDPGLog.h [hunk 5]
```diff
@@ -72,7 +72,7 @@ class MOSDPGLog : public MessageInstance<MOSDPGLog, MOSDPeeringOp> {
     set_priority(CEPH_MSG_PRIO_HIGH); 
   }
   MOSDPGLog(shard_id_t to, shard_id_t from,
-	    version_t mv, pg_info_t& i, epoch_t query_epoch)
+	    version_t mv, const pg_info_t& i, epoch_t query_epoch)
```
Comment (rzarzynski) [inline src/messages/MOSDPGLog.h]: OK.

File: src/osd/osd_types.h [hunk 6]
```diff
@@ -133,9 +133,6 @@ struct osd_reqid_t {
   osd_reqid_t()
     : tid(0), inc(0)
   {}
-  osd_reqid_t(const osd_reqid_t& other)
```
Comment (rzarzynski) [inline src/osd/osd_types.h]: OK.

File: src/messages/MOSDPGInfo.h [hunk 7]
```diff
@@ -29,17 +29,22 @@ class MOSDPGInfo : public MessageInstance<MOSDPGInfo> {
...
 public:
-  vector<pair<pg_notify_t,PastIntervals> > pg_list;
+  using pg_list_t = std::vector<std::pair<pg_notify_t,PastIntervals>>;
+  pg_list_t pg_list;
 
   epoch_t get_epoch() const { return epoch; }
 
   MOSDPGInfo()
-    : MessageInstance(MSG_OSD_PG_INFO, HEAD_VERSION, COMPAT_VERSION) {
-    set_priority(CEPH_MSG_PRIO_HIGH);
```
[the remark this answers is not in the document]
Comment (tchaikov) [reply src/messages/MOSDPGInfo.h]: @rzarzynski i am following you here. i think the original behavior is preserved with this change.

File: src/osd/OSD.cc (hunk 2 above)
Comment (tchaikov) [reply src/osd/OSD.cc]: yeah, the current semantic is also easier to understand. the notify_list is supposed to be consumed by `do_notifies`.

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2019-03-20_05:45:48-rados-wip-kefu-testing-2019-03-20-1120-distro-basic-smithi/

**Pull request merged**